src/context/cart-context.tsx

```diff
@@ -32,13 +33,15 @@ const CartContext = createContext({} as CartContextInterface);
 
 const CartProvider = ({ children }: PropsContextModel) => {
   const [cart, setCart] = useState<CartModel | undefined>(undefined);
+
+  const [handleButton, setHandleButton] = useState<boolean>(false);
```

Comment:
Bad name. It is not button whatsoever. It is a flag of ... ? What represents this flag?